### ISSUE TITLE
mstpr-brainbot - Inconsistency in healing between docs and code, healing is only possible after 2 rounds

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-10-looksrare-judging#8]
mstpr-brainbot

high

# Inconsistency in healing between docs and code, healing is only possible after 2 rounds
## Summary
The documentation states that healing is possible after 1 round, as evidenced here:
"So, for example, if an agent is wounded in round 12, the first round they can heal is round 13, denoted as EligibleRoundCount = 1."
https://github.com/sherlock-audit/2023-10-looksrare/blob/main/contracts-infiltration/README.md#percentage-chance-to-heal

However, the code implementation permits healing only after 2 rounds. Therefore, the correct statement should be: "The first round they can heal is round 14, not 13."
## Vulnerability Detail
As we can see in the following link that the heal is only possible after 2 rounds, assuming an agent is wounded at round 13, earliest possible round that the agent can heal hisself is the round 15 due to this check here:
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L843-L847

Assuming the agent is wounded at round 13 and the current round is 14:
13 - 14 < 2 = 1 < 2 = true = revert.

To give more details, Assuming that the active agents are higher than the secondary wins count for at least 3 more rounds, if the game is in round 13 in order for game to be in round 14 the chainlink VFR needs to call the fullFill function. Once the fulfill function is called the current rounds wounded agents will be determined and the round id will increase to 14. Once the game is in round 14 now we need to wait for a day and call startNewRound this will trigger the VFR and now we can expect an another fulfill call from VFR and the round will be updated to 15. Once the round is 15 now the wounded agent in round 13 can call heal because 15-13 < 2 = 2 < 2 = false so there will be no reverts and function can work. 

## Impact
This is clearly not consistent with the README of the page so I am assuming this is wrongly implemented hence, I will classify this as high 

## Code Snippet
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L843-L847
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L1096-L1196
https://github.com/sherlock-audit/2023-10-looksrare/blob/86e8a3a6d7880af0dc2ca03bf3eb31bc0a10a552/contracts-infiltration/contracts/Infiltration.sol#L579-L651
## Tool used

Manual Review

## Recommendation
Either change the doc accordingly or make the if statement as follows:
```solidity
unchecked {
                if (currentRoundId - woundedAt < 1) {
                    revert HealingMustWaitAtLeastOneRound();
                }
            }
```

Duplicate of #44
